velox/exec/prefixsort/PrefixSortEncoder.h

```diff
@@ -294,9 +294,9 @@ FOLLY_ALWAYS_INLINE void PrefixSortEncoder::encodeNoNulls(
   } else {
     // 'data' is stored in non-contiguous allocation pieces in the row
     // container, we only read prefix size data out.
-    auto stream = HashStringAllocator::prepareRead(
+    HashStringAllocator::InputStream stream(
         HashStringAllocator::headerOf(value.data()));
-    stream->readBytes(dest, copySize);
+    stream.ByteInputStream::readBytes(dest, copySize);
```

Comment:
You have to add this prefix because the InputStream class was forward declared inside the HashStringAllocator, right?

Could you inline the definition in side of HashStringAllocator, or define it outside the class in the same header? HashStringAllocatorInputStream would be just as clear as HashStringAllocator::InputStream, slightly shorter, and then we don't need all these ByteInputStream:: prefixes.